src/time.cc

```diff
@@ -57,17 +62,36 @@ std::string ToString(const std::chrono::system_clock::time_point& t) {
 
 std::chrono::system_clock::time_point FromString(const std::string& s) {
   std::tm tm;
-  const char* end = strptime(s.c_str(), "%Y-%m-%dT%H:%M:%S", &tm);
-  if (end == nullptr || end - s.c_str() != s.find('.')) {
-    // TODO
+  char* const end = strptime(s.c_str(), "%Y-%m-%dT%H:%M:%S", &tm);
+  if (end == nullptr || !std::isdigit(*(end - 2))) {
+    // TODO: Invalid time format.
     return std::chrono::system_clock::time_point();
   }
   char* zone;
-  long ns = std::strtol(end + 1, &zone, 10);
-  if (zone <= end + 1 || *zone != 'Z' || *(zone+1) != '\0') {
-    // TODO
+  if (*end == '.') {
+    (void)std::strtol(end + 1, &zone, 10);
+    if (zone <= end + 1) {
+      // TODO: Missing nanoseconds.
+      return std::chrono::system_clock::time_point();
```

Comment:
Is it not safe to assume 0 for nanoseconds?

Reply:
It depends on whether we treat a trailing decimal point with no digits after as valid input or an error. As I read the [spec](http://www.faqs.org/rfcs/rfc3339.html), a decimal point must be followed by at least one digit (search for `time-fraction`).